toponetx/classes/cell_complex.py

```diff
@@ -1793,7 +1826,15 @@ def singletons(self):
         """
         return [node for node in self.nodes if self.degree(node) == 0]
 
-    def remove_singletons(self, name=None):
+    def clone(self) -> "CellComplex":
+        """Create a clone of the CellComplex."""
+        _G = self._G.copy()
+        CX = CellComplex(_G, self.name)
+        for cell in self.cells:
+            CX.add_cell(cell)
+        return CX
+
+    def remove_singletons(self, name: Optional[str] = None) -> "CellComplex":
```

Comment:
since the context is missing here:

The documentation said 'creates a clone with singleton nodes removed', but the implementation modified `self` by removing said nodes.

Reply:
Currently, most methods mutate the object so it feels weird that this method in particular returns a new instance. Is there a particular reason that this method is special? Otherwise I would strive for consistency.

Reply:
It is consistent with NetworkX' `subgraph` or our `restrict_to_{nodes,cells}` methods, which also return a copy.
Still, I think you're probably right since it is most similar to `remove_equivalent_cells` which instead modifies it.

I'll revert to the old behavior and update the docstring instead.

Reply:
I think there is a distinction between `remove` and `restrict`. For the later I would expect a new object, for the former not necessarily.